src/Compilers/CSharp/Test/Semantic/Semantics/NullConditionalAssignmentTests.cs

```diff
@@ -2892,5 +2892,460 @@ static void M(bool b, C? c1, C? c2)
                 """,
                 graph, symbol);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/7502")]
+        public void PointerReturnType_Simple()
+        {
+            var source = """
+                public unsafe class A
+                {
+                    public byte* Ptr = null;
+                }
+
+                unsafe class Test
+                {
+                    static void M1(A a)
+                    {
+                        byte* ptr = a?.Ptr;
+                    }
+
+                    static void M2(A a)
+                    {
+                        var result = a?.Ptr;
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/7502")]
+        public void PointerReturnType_WithUsage()
+        {
+            var source = """
+                using System;
+
+                public unsafe class A
+                {
+                    public byte* Ptr;
+                }
+
+                unsafe class Test
+                {
+                    static void Main()
+                    {
+                        byte b = 42;
+                        var a = new A { Ptr = &b };
+                        byte* ptr1 = a?.Ptr;
+                        Console.Write(ptr1 == null ? "null" : *ptr1);
+
+                        a = null;
+                        byte* ptr2 = a?.Ptr;
+                        Console.Write(ptr2 == null ? " null" : " not null");
+                    }
+                }
+                """;
+            var verifier = CompileAndVerify(source, options: TestOptions.UnsafeDebugExe, verify: Verification.Skipped, expectedOutput: "42 null");
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("Test.Main", """
```

Comment:
We only need to verify the IL for one of the cases. This is redundant and harder to review